### DISCUSSION TITLE
Unable to use Editable inside array to update values

### DISCUSSION THREAD

**Discussion by spidertwin2** [chakra-ui/chakra-ui#6093] [Q&A] [answered]
Hello,

I am using`Editable` to render a list of questions.  But I am unable to update the value `onChange` correctly.

Plus, `useEffect` is not firing questions to console when values are changed.

```
const QuizQuestions = () => {
  const [questions, setQuestions] = useState([
    "How many colors in a rainbow?",
    "How many corners in a pizza?"
  ]);

  const handleChange = (val, i) => {
    let tempQuestions = questions;
    tempQuestions[i] = val;
    setQuestions(tempQuestions);
  };

  useEffect(() => {
    console.log("questions: ", questions);
  }, [questions]);

  return (
    <Box>
      {questions.map((question, i) => (
        <Editable
          defaultValue={question}
          onChange={(val, i) => handleChange(val, i)}
        >
          <EditablePreview />
          <EditableInput />
        </Editable>
      ))}
    </Box>
  );
};
```

Sandbox [link](https://codesandbox.io/s/reverent-hamilton-fvqci6?file=/src/QuizQuestions.tsx).

Thanks thanks 🎋

**Comment by hugobqd** [comment, marked as answer]
Hello!

- First you have to make a temporary copy of your questions's array in _handleChange_ : `let tempQuestions = [...questions]`.
- Then,  the _onChange_ only return the new value, so remove `i` from params : `onChange={(val) => handleChange(val, i)}`.
- Also don't forget to give an unique key for map's items `key={i}`.

And then it work's fine! 🎋

[Modified sandbox link](https://codesandbox.io/s/jovial-dubinsky-iifko3?file=/src/QuizQuestions.tsx)

**Reply by spidertwin2** [reply]
Merci bien @hugobqd 🪴
